repairnator/repairnator-scanner/src/main/java/fr/inria/spirals/repairnator/scanner/ScannerBuildListener.java

```diff
@@ -0,0 +1,32 @@
+package fr.inria.spirals.repairnator.scanner;
+
+import fr.inria.spirals.repairnator.BuildListener;
+import javax.jms.MessageListener;
+import javax.jms.Message;
+
+public class ScannerBuildListener implements BuildListener,MessageListener{
+    private static ScannerBuildListener scannerBuildListener;
+    private static Launcher launcher;
+
+    public ScannerBuildListener (){}
+    public void setLauncher (Launcher launcher) {
+        this.launcher = launcher;
+    }
+
+    public static ScannerBuildListener getInstance() {
+        if (scannerBuildListener == null) {
+            scannerBuildListener = new ScannerBuildListener();
+        }
+        return scannerBuildListener;
+    }
+    public void runAsConsumerServer() {}
```

Comment:
If every implementation needs this method and the `setLauncher`, they should be put into the `BuildListener` interface.